### PR TITLE
Test docker-storage-setup --reset for DOCKER_ROOT_VOLUME=yes

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Signed-off-by: Shishir Mahajan <shishir.mahajan@redhat.com>

### CODE REVIEW

**Comment by shishir-a412ed** [comment]
ping @rhvgoyal

File: tests/011-test-docker-root-volume-cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+ # Test failed.
+ if [ $? -ne 0 ]; then
+    echo "ERROR: $testname: $DSSBIN Failed." >> $LOGS
+    cleanup $vg_name "$devs"
+    return $test_status
+ fi
+
+  $DSSBIN --reset >> $LOGS 2>&1
+  # Test failed.
```
Comment (rhvgoyal) [inline tests/011-test-docker-root-volume-cleanup.sh]: Why don't you simply return from here if DSSBIN failed and returned non-zero exit code. Just that we will have to do cleanup before we return.

File: tests/011-test-docker-root-volume-cleanup.sh [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+     if lv_exists "$vg_name" "$docker_root_lv_name";then
+          test_status=1
+     fi
+  fi
+  if [ ${test_status} -eq 1 ]; then
+     echo "ERROR: $testname: $DSSBIN --reset Failed." >> $LOGS
+  fi
+
+  cleanup $vg_name "$devs"
+  return $test_status
```
Comment (rhvgoyal) [inline tests/011-test-docker-root-volume-cleanup.sh]: if dss --reset failed, then we have to do the cleanup ourselves. That is remove docker-root-lv, unmount, remove mount unit file etc.

File: tests/011-test-docker-root-volume-cleanup.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  $DSSBIN --reset >> $LOGS 2>&1
+  # Test failed.
+  if [ $? -eq 0 ]; then
+     test_status=0
+     if [ -e "/etc/sysconfig/docker-storage" ] || [ -e "/etc/systemd/system/${mount_filename}" ]; then
+          test_status=1
+     fi
+     if lv_exists "$vg_name" "$docker_root_lv_name";then
+          test_status=1
+     fi
```
Comment (rhvgoyal) [inline tests/011-test-docker-root-volume-cleanup.sh]: I think I like to check for error condition first and if there is error, do the cleanup and return non-zero exit code. Something like.

if lv_exists; then
   cleanup_all()
   return $test_status
fi

Here cleanup_all() will cleanup docker_root_lv as well as call cleanup().

Comment (shishir-a412ed) [inline tests/011-test-docker-root-volume-cleanup.sh]: I think we can reuse `cleanup_container_root_volume` (Rename that to cleanup_all() ).
https://github.com/projectatomic/docker-storage-setup/blob/master/tests/010-test-use-devs-to-create-docker-root-volume.sh#L57-L67 if we move this to `libtest.sh`.
It does the following:
1) Unmount the mount path.
2) Cleanup `docker-root-lv`.
3) Cleanup the mount file.

**Comment by shishir-a412ed** [comment]
@rhvgoyal PTAL.

File: tests/011-test-docker-root-volume-cleanup.sh [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  $DSSBIN --reset >> $LOGS 2>&1
+  # Test failed.
+  if [ $? -ne 0 ]; then
+    echo "ERROR: $testname: $DSSBIN --reset Failed." >> $LOGS
+    cleanup_all $vg_name $docker_root_lv_name $mount_path "$devs"
+    return $test_status
+  fi
+
+  if ! everything_clean $vg_name $docker_root_lv_name $mount_filename;then
+    echo "ERROR: $testname: $DSSBIN --reset Failed." >> $LOGS
```
Comment (rhvgoyal) [inline tests/011-test-docker-root-volume-cleanup.sh]: This error message probly needs to be changed now. say.

`ERROR: $testname: reset did not cleanup everything as needed.`

Comment (shishir-a412ed) [inline tests/011-test-docker-root-volume-cleanup.sh]: Fixed.

**Comment by shishir-a412ed** [comment]
@rhvgoyal PTAL.

File: tests/011-test-docker-root-volume-cleanup.sh [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    cleanup_all $vg_name $docker_root_lv_name $mount_path "$devs"
+    return $test_status
+  fi
+
+  return 0
+}
+
+everything_clean(){
+local vg_name=$1
+local docker_root_lv_name=$2
```
Comment (rhvgoyal) [inline tests/011-test-docker-root-volume-cleanup.sh]: spacing is not proper in this function in all the lines.

File: tests/011-test-docker-root-volume-cleanup.sh [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+  umount $mount_path >> $LOGS 2>&1
+  lvchange -an $vg_name/${lv_name} >> $LOGS 2>&1
+  lvremove $vg_name/${lv_name} >> $LOGS 2>&1
+
+  cleanup_mount_file $mount_path
+  cleanup $vg_name "$devs"
+}
+
+#  If a user has specified DOCKER_ROOT_VOLUME=yes
```
Comment (rhvgoyal) [inline tests/011-test-docker-root-volume-cleanup.sh]: There is no need of space after `#`

**Comment by rhvgoyal** [comment]
I ran the tests and after that system is not clean. There is so much left behind. (despite the fact that all tests passed).

**Comment by rhvgoyal** [comment]
I think we will have to call cleanup() even if test is successful. And reason being that dss --reset does not cleanup volume group, physical volumes, partitions and /etc/sysconfig/docker-storage-setup file.

**Comment by shishir-a412ed** [comment]
Right, calling `cleanup()` after everything is successful would be required. I will add the cleanup call.

**Comment by shishir-a412ed** [comment]
@rhvgoyal Please check now.

File: tests/011-test-docker-root-volume-cleanup.sh [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+everything_clean(){
+  local vg_name=$1
+  local docker_root_lv_name=$2
+  local mount_filename=$3
+  if [ -e "/etc/sysconfig/docker-storage" ] || [ -e "/etc/systemd/system/${mount_filename}" ]; then
+    return 1
+  fi
+  if lv_exists "$vg_name" "$docker_root_lv_name";then
```
Comment (rhvgoyal) [inline tests/011-test-docker-root-volume-cleanup.sh]: This should be ` ! lv_exists` ?

Comment (rhvgoyal) [inline tests/011-test-docker-root-volume-cleanup.sh]: may be not. ignore this comment.

**Comment by rhvgoyal** [comment]
LGTM

**Comment by rhvgoyal** [comment]
@rh-atomic-bot r+ deaf03f

**Pull request closed without merging**